x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/execution/ExecutionService.java

```diff
@@ -427,16 +432,23 @@ private void logWatchRecord(WatchExecutionContext ctx, Exception e) {
     }
 
     /*
-       The execution of an watch is split into two phases:
+       The execution of a watch is split into two phases:
        1. the trigger part which just makes sure to store the associated watch record in the history
        2. the actual processing of the watch
        The reason this split is that we don't want to lose the fact watch was triggered. This way, even if the
        thread pool that executes the watches is completely busy, we don't lose the fact that the watch was
        triggered (it'll have its history record)
     */
     private void executeAsync(WatchExecutionContext ctx, final TriggeredWatch triggeredWatch) {
+        final ThreadContext threadContext = clusterService.threadPool().getThreadContext();
+        this.tracer.startTrace(threadContext, SpanId.forBareString(""), "", Map.of());
+
         try {
-            executor.execute(new WatchExecutionTask(ctx, () -> execute(ctx)));
+            executor.startTrace(ctx.id());
```

Comment:
Does `ctx.id()` then translates to the unique ID each watch has e.g. `2aef85...`? Will I see each watch as a unique transaction in the APM view with the ID being the transaction indicator? If so, we should probably prefix that with `Watcher: id`.